src/Util/FisherYates/Interface.dfy

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module FisherYates.Interface {
+  import Uniform
+
+  trait {:termination false} Trait extends Uniform.Interface.Trait {
+
+    method Shuffle<T(!new)>(a: array<T>)
+      decreases *
```

Comment:
Why decreases *? We should be able to prove termination for this one.

Reply:
As discussed offline, the `decreases *` is inherited from `Uniform`.